### PR TITLE
EOS-20937: SSPL: Use provisioner supported server and storage type in node health

### PR DESCRIPTION
**Pull request by mariyappanp**
## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [ ] Unittest updated and all unittests are passing?

* [ ] Sanity/Self tests are updated (if applicable)?

* [ ] Manual testing done covering happy path and non-happy path?

## Integration:

* [x] If there is any interface change, did you communicate it to other components?
* [x] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [x] Did you complete deployment test if applicable?

## PR checklist:
* [ ] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py [hunk 1]
```diff
@@ -782,4 +785,4 @@ def get_psus():
                 key, value = stack.pop().strip().split(":")
                 psu[key] = value.strip()
             psus.append(psu)
-        return psus
+        return psus
```
Comment (nitin-seagate) [inline low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py:788]: Add new line

**Review by jjxsg — APPROVED** [review]

**Review by thavanathan — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py [hunk 2]
```diff
@@ -87,12 +87,15 @@ def __init__(self):
 
     def validate_server_type_support(self):
         """Check for supported server type."""
-        server_type = Conf.get(GLOBAL_CONF, NODE_TYPE_KEY).lower()
-        # TODO Add support for 'virtual' type server.
+        server_type = Conf.get(GLOBAL_CONF, NODE_TYPE_KEY)
```
Comment (nitin-seagate) [inline low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py:90]: Provisioner supports following server types:
`VM, HW`

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py [hunk 3]
```diff
@@ -65,12 +65,15 @@ def __init__(self):
 
     def validate_storage_type_support(self):
         """Check for supported storage type."""
-        storage_type = Conf.get(
-            GLOBAL_CONF, STORAGE_TYPE_KEY, "virtual").lower()
+        storage_type = Conf.get(GLOBAL_CONF, STORAGE_TYPE_KEY)
```
Comment (nitin-seagate) [inline low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py:68]: Just FYI, Provisioner supports following storage types:
`RBOD, JBOD, EBOD, virtual`

**Review by nitin-seagate — APPROVED** [review]
Looks good to me, in-line with values provided by Provisioner

**Review by nitin-seagate — APPROVED** [review]
Looks good to me, in-line with values provided by Provisioner

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py (hunk 1 above)
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py]: Done.

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py (hunk 2 above)
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py:90]: Yes, covered the same.

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py (hunk 3 above)
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py:68]: Okay. 
As per design, resource health generation is supported for all the above except virtual storage type.

**Review by satish-seagate — APPROVED** [review]

**Review by gargpalak048 — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py [hunk 4]
```diff
@@ -65,12 +65,15 @@ def __init__(self):
...
         logger.debug(self.log.svc_log(f"Storage Type:{storage_type}"))
-        supported_types = ["5u84", "rbod", "pods", "corvault"]
-        if storage_type not in supported_types:
-            msg = f"Health provider is not supported for storage type {storage_type}"
+        if not storage_type:
+            msg = "ConfigError: storage type is unknown."
+            logger.error(self.log.svc_log(msg))
+            raise ResourceMapError(errno.EINVAL, msg)
+        supported_types = ["rbod", "jbod", "ebod"]
```
[the remark this answers is not in the document]
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py]: Addressed. Please review.

[the remark this answers is not in the document]
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py]: Done.

**Review by sandeepsgt — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**